### ISSUE TITLE
[Webchat/Emulator][Customer Reported] If a CardAction of type MessageBack doesn't have the  value property initialized the button is not clickable

### ISSUE THREAD

**Issue by p-nagpal** [microsoft/BotFramework-WebChat#2956]
## Version

4.6.0 but also reproduces with the version in latest emulator.

## Describe the bug

A customer has reported an issue in WebChat/Emulator, where if a CardAction of type MessageBack doesn't have the  value property initialized (value is null) the button is not clickable (no action when clicked). Since some channels do not expect object types (such as Facebook), we encourage developers to use the text property instead of value. Note that text is string and value is of object type. If they intent to pass object type  data then they just use channelData. We have been able to repro this issue. Other channels such as Facebook and MSTeams work with text, even when value is null. This fix is needed so that developers can have the similar experience across channels.
 
spec: https://github.com/microsoft/botframework-sdk/blob/master/specs/botframework-activity/botframework-activity.md#message-back

[Bug]

**Comment by corinagum** [comment]
I'm confused. The spec says that value may be of any type. Does sending a string in the value not work?

**Comment by p-nagpal** [comment]
Hi @corinagum, thank you for help. We found that sending in a string in the value works. However sending in a 'null' or uninitialized value does not (the button is not clickable).

**Comment by compulim** [comment]
We tried in Web Chat with:
- `CardAction` in suggested actions
- `text` set to a string
- `value` is null

And it works.

Repro:
- Type `suggested-actions` with MockBot
- Click the very last suggested action, named "Message back as string with display text"

![image](https://user-images.githubusercontent.com/1622400/75507060-d1e99700-5994-11ea-9a60-34a39cc4d024.png)

![image](https://user-images.githubusercontent.com/1622400/75507053-c8f8c580-5994-11ea-9851-57e8ae974ec6.png)

**Comment by corinagum** [comment]
@p-nagpal please provide customer code.

**Comment by bwamie** [comment]
Fails if you send it as an attachment (HeroCard button)

```
var actions = new List<CardAction>()
                {
                    new CardAction() { Title = "Red", Type = ActionTypes.ImBack, Value = "Red" },
                    new CardAction() { Title = "Yellow", Type = ActionTypes.ImBack, Value = "Yellow" },
                    new CardAction() { Title = "DeadBlueButton", Type = ActionTypes.MessageBack, Text = "DeadBlueButton", Value = null },
                    new CardAction(ActionTypes.MessageBack, "MessageBack", null, "This is a MessageBack.", displayText: "This is a displayText.", value: "{'JsonValue': 'MessageBackValue'}")
                };

            reply.Attachments.Add(new Attachment()
            {
                ContentType = HeroCard.ContentType,
                Content = new HeroCard()
                {
                    Title = "My HeroCard",
                    Buttons = actions.Where(a => a != null).ToList()
                }
            });
```

**Comment by compulim** [comment]
## Screenshot

![image](https://user-images.githubusercontent.com/1622400/75589415-8a641900-5a2f-11ea-9cc5-89edc9d5ded8.png)

The bug is in this code, it didn't send `text` field.

## Technical background

Hero card is rendered thru `AdaptiveCardBuilder`, which turn into an Adaptive Card for visualization only (not logic, i.e. not submit).

For data submission, since AC only support `Action.Submit` but not wide variety format of Direct Line (`imback`, `postback`, `messageback`), thus, there is a conversion we done.

The conversion is the left screenshot above, it lacked of `text` field.

When the user click on the hero card button, it is sent thru AC as an `Action.Submit`. Our code at `AdaptiveCardRenderer` intercepted the `Action.Submit`, inspect it, and turn it into `imback`, `postback`, and `messageback`. That is the right screenshot above.

## How to fix

1. Fix the screenshot above but adding `text`
1. Add a new card on MockBot that have 6 buttons
   - Refer to `suggested-actions` command for those 6 combinations (`imback`, `postback`, `messageback`)
1. Add a new test that click each of the 6 buttons
   - The expected outcome should be exactly as those on suggested-actions

**Comment by compulim** [comment]
This is fixed in #3003 by @geea-develop.
